### PR TITLE
add curriculumbuilder tip support

### PR DESCRIPTION
**Pull request by Hamms**
As defined in https://github.com/mrjoshida/curriculumbuilder/blob/master/curriculumBuilder/tips.py

### CODE REVIEW

**Comment by Hamms** [comment]
Note that when reviewing, it's probably best to ignore the diff of `plugins/tip.js` and just look at the [final version of the file](https://github.com/code-dot-org/cdo-flavored-markdown/blob/d088de92a6b2ad71c9e338229b8645e67f1a4b73/src/plugins/tip.js) (since the content that was originally in that file was entirely random placeholder garbage)

File: src/plugins/compiler/indent.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+/**
+ * Stringify text without escaping special characters; useful for rendering
+ * custom syntaxes which include control characters back to markdown.
+ *
+ * @see https://github.com/remarkjs/remark/blob/remark-stringify%405.0.0/packages/remark-stringify/lib/visitors/text.js
+ * @see divclass
```
Comment (davidsbailey) [inline src/plugins/compiler/indent.js]: unclear how these `@see` tags are related to the current file. copy-pasta?

Comment (Hamms) [reply src/plugins/compiler/indent.js]: Dangit, yes; my bad, I'll clean this up

File: src/plugins/parser/tip.js [hunk 2]
```diff
@@ -1,34 +1,133 @@
...
-  const match = /^@(\w+)/.exec(value);
+  if (silent) {
+    return true;
+  }
 
-  if (match) {
-    if (silent) {
-      return true;
+  // find the indented block that represents the content of the tip. Blocks can
+  // be indented by either four spaces or a tab character
```
Comment (davidsbailey) [inline src/plugins/parser/tip.js]: might be helpful to add: `// Blocks can also contain blank lines separating multiple indented paragraphs.`

**Pull request merged**